plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java

```diff
@@ -116,6 +116,8 @@ public static ClientConfiguration buildConfiguration(Logger logger, Settings set
             AwsSigner.configureSigner(awsSigner, clientConfiguration, endpoint);
         }
 
+        clientConfiguration.setSocketTimeout((int) CLOUD_S3.READ_TIMEOUT.get(settings).millis());
```

Comment:
I know that @tlrx prefers when we use the AWS SDK defaults instead of fixing ourselves a setting. And I agree.

Can we write instead something like:

```java
if (READ_TIMEOUT.exists(settings) || CLOUD_S3.READ_TIMEOUT.exists(settings)) {
    clientConfiguration.setSocketTimeout((int) CLOUD_S3.READ_TIMEOUT.get(settings).millis());
}
```

And then remove in doc that there is a default value for this setting?